src/views/Login.js

```diff
@@ -0,0 +1,163 @@
+import React, { Component } from "react";
+import PropTypes from "prop-types";
+import { connect } from "react-redux";
+import { handleTextInput, handleSignIn } from "../redux/actions/loginActions";
+import { isEmpty } from "../utils/helperFunctions";
+import Validator from "../utils/validator";
+import TextInput from "../components/common/Inputs/TextInput";
+import FormButton from "../components/common/Buttons/FormButton";
+import BasicButton from "../components/common/Buttons/BasicButton";
+import SocialButton from "../components/common/Buttons/SocialButton";
+
+export class Login extends Component {
+  constructor(props) {
+    super(props);
+    this.state = {
```

Comment:
Why using a local state here?
Is it possible to keep all the component state in redux?

Reply:
these kind of data are temporary, and I will not need them in other components. So, for the sake of simplicity and keeping a maintainable codebase, I handled them inside the component.